iterator_test.go

```diff
@@ -2737,3 +2738,179 @@ func BenchmarkSeekPrefixTombstones(b *testing.B) {
 		iter.SeekPrefixGE(seekKey)
 	}
 }
+
+// BenchmarkQueueWorkload benchmarks a workload consisting of multiple queues
+// that are all being processed at the same time. Processing a queue entails either
+// appending to the end of the queue (a Set operation) or deleting from the start of
+// the queue (a Delete operation). The goal is to detect cases where we see a large
+// buildup of point tombstones at the beginning of each queue, which leads to the
+// slowdown of SeekGE(<start of queue>). To that end, the test progresses through
+// a series of steps that each 1) process the queues a certain number of times
+// and then 2) benchmark both the queue processing throughput and SeekGE performance.
+// See https://github.com/facebook/rocksdb/wiki/Implement-Queue-Service-Using-RocksDB
+// for more information.
+func BenchmarkQueueWorkload(b *testing.B) {
+	const queueCount = 25
+	const minValSize = 512
```

Comment:
I tried this initially, but increasingly found that this runs into the same issue of lack of comparability/interpretability that comes with randomizing the delete ratios. I opted to go without the randomization entirely and use a slice of value sizes. It's set to just `{128, 2048}` since that seems to be enough to show the difference that value size creates (i.e. tombstone buildup is worse with larger value sizes), but we can always run this with other value sizes in the future too.